src/CommandHandler/RefundUnitsHandler.php

```diff
@@ -21,34 +20,32 @@
     /** @var RefunderInterface */
     private $orderShipmentsRefunder;
 
-    /** @var OrderRefundingAvailabilityCheckerInterface */
-    private $orderRefundingAvailabilityChecker;
-
     /** @var EventBus */
     private $eventBus;
 
     /** @var OrderRepositoryInterface */
     private $orderRepository;
 
+    /** @var RefundUnitsCommandValidatorInterface */
+    private $refundUnitsCommandValidator;
+
     public function __construct(
         RefunderInterface $orderUnitsRefunder,
         RefunderInterface $orderShipmentsRefunder,
-        OrderRefundingAvailabilityCheckerInterface $orderRefundingAvailabilityChecker,
         EventBus $eventBus,
-        OrderRepositoryInterface $orderRepository
+        OrderRepositoryInterface $orderRepository,
+        RefundUnitsCommandValidatorInterface $refundUnitsCommandValidator
     ) {
-        $this->orderRefundingAvailabilityChecker = $orderRefundingAvailabilityChecker;
         $this->orderUnitsRefunder = $orderUnitsRefunder;
         $this->orderShipmentsRefunder = $orderShipmentsRefunder;
         $this->eventBus = $eventBus;
         $this->orderRepository = $orderRepository;
+        $this->refundUnitsCommandValidator = $refundUnitsCommandValidator;
     }
 
     public function __invoke(RefundUnits $command): void
     {
-        if (!$this->orderRefundingAvailabilityChecker->__invoke($command->orderNumber())) {
-            throw OrderNotAvailableForRefundingException::withOrderNumber($command->orderNumber());
-        }
+        $this->refundUnitsCommandValidator->validate($command);
```

Comment:
I'm wondering should it be placed here or maybe it should be validated during command creation (see [this comment](https://github.com/Sylius/RefundPlugin/pull/62#pullrequestreview-156230719))? But it can be for sure changed after the merge ;)